Source/Dafny/Resolver.cs

```diff
@@ -12501,6 +12494,48 @@ private class RBranchExpr : RBranch {
       }
     }
 
+    private ExtendedPattern RemoveDisjunctivePatterns(ExtendedPattern pat, bool allowVariables) {
+      switch (pat) {
+        case LitPattern:
+          return pat;
+        case IdPattern p:
+          if (p.Arguments == null && !p.Id.StartsWith("_") && !allowVariables) {
```

Comment:
What is the check `!p.Id.StartsWith("_")` for?

Reply:
Wildcards (`_` patterns) are implemented as `_`-prefixed variables, and we don't want to complain about those.

Reply:
From the documentation:

```
WildIdent = NoUSIdent | "_"
Identifier, disallowing leading underscores, except the “wildcard” identifier _. When _ appears it is replaced by a unique generated identifier distinct from user identifiers. This wildcard has several uses in the language, but it is not used as part of expressions.
```

Shouldn't it be `p.Id == "_"` then?

Also, is wildcard a good name? Hole seems more fitting.

Reply:
> Shouldn't it be p.Id == "_" then?  If the point of a wildcard is to effectively not capture the variable, then why would we permit an identifier after the underscore?

No: internally, "_" is treated as a variable and gets renamed into a unique variable name starting with an underscore, so by the time we get here the holes are named something like `_v64`.

A future refactoring should probably just add a `WildcardPattern` (or `HolePattern`, indeed; I think it's a better fit)

> Also, shouldn't there be a function Identifier.IsWildcard(p.id) to make this more explicit?

OK, I will do that.  Even better, I can make it a property of the identifier pattern.

Reply:
Done